server/src/main/java/org/elasticsearch/common/geo/GeometryTreeReader.java

```diff
@@ -48,20 +48,28 @@ public Extent getExtent() throws IOException {
         }
         assert input.readVInt() == 1;
         ShapeType shapeType = input.readEnum(ShapeType.class);
-        if (ShapeType.POLYGON.equals(shapeType)) {
-            EdgeTreeReader reader = new EdgeTreeReader(input);
-            return reader.getExtent();
-        } else {
-            throw new UnsupportedOperationException("only polygons supported -- TODO");
+        switch (shapeType) {
```

Comment:
I'll migrate this out into a helper function within the geometrytreereader. Since that is the only place that uses this, I'd like to keep it lighter than a Class